File: 2019/day01-1.hs
import System.IO

solution :: String -> Int
solution content = foldr (+) 0 . map (\x -> x - 2) . map (\x -> div x 3) . map (read) . lines $ content

main :: IO ()
main = do
    handle <- openFile "day01.dat" ReadMode
    content <- hGetContents handle
    print . solution $ content

File: 2019/day01-1.ts
function solution(content: string) {
  return content
    .split("\n")
    .map((item) => Math.floor(parseInt(item) / 3) - 2)
    .reduce((acc, cur) => acc + cur, 0);
}

Deno.readTextFile("day01.dat")
  .then((content) => content.trim())
  .then(solution)
  .then(console.log);
